x-pack/plugins/lens/public/embeddable/interfaces/has_lens_config.ts

```diff
@@ -5,14 +5,25 @@
  * 2.0.
  */
 
-import { type HasType, apiIsOfType } from '@kbn/presentation-publishing';
+import type {
+  HasParentApi,
+  HasType,
+  PublishesLocalUnifiedSearch,
+  PublishesPanelTitle,
+} from '@kbn/presentation-publishing';
+import { apiIsOfType } from '@kbn/presentation-publishing';
 import { LensSavedObjectAttributes } from '../embeddable';
 
 export type HasLensConfig = HasType<'lens'> & {
   getSavedVis: () => Readonly<LensSavedObjectAttributes | undefined>;
 };
 
-export const apiHasLensConfig = (api: unknown): api is HasLensConfig => {
+export type LensApi = HasLensConfig &
+  PublishesPanelTitle &
+  PublishesLocalUnifiedSearch &
+  Partial<HasParentApi<unknown>>;
+
+export const apiHasLensConfig = (api: unknown): api is LensApi => {
```

Comment:
This type guard isn't accurate anymore. It's returning true if the given api is a `LensApi`, but it isn't checking for the presence of a local unified search or a panel title, both of which are required.

I recognize that currently the `apiIsOfType` check precludes all other APIs from passing this type guard, but there is a shape that would pass the type guard and cause a type error down the line.

I'd also rename the type guard to `isLensApi` or something similar.

Reply:
resolved with https://github.com/elastic/kibana/pull/176869/commits/835e3ae067735d90ce9cc3c9651e1b924ec79bfd